### ISSUE TITLE
mautrix.errors.crypto.SessionNotFound: Messages on Matrix are not bridged to Telegram

### ISSUE THREAD

**Issue by Craeckie** [tulir/mautrix-telegram#611]
Hi,
first of all thanks for this bridge and all the effort you put into it!

I just encountered this error when trying to send a message in a private encrypted chat. 

The chat was added and backfilled when I received a message from this person. Afterwards I sent a message in Matrix and didn't notice that it wasn't bridged. That person sent me another message which was successfully bridged. When I wrote another message in Matrix it wasn't bridged either and then I noticed the error in the log. Messages I send on Telegram are still bridged to Matrix, just not vice-versa. The other unencrypted chats still work.

I just updated to the latest commit 78fcacf7aa5bb97a85df4b8dcabf34bd7e54c686 and I'm still not able to send messages
Some time ago I switched from SQLite to postgres, not sure if it has something to do with this error.

<pre>
telegram-bridge_1  | [2021-04-12 12:30:29,812] [DEBUG@mau.bridge.e2ee] Didn't find session &lt;session id&gt;, waiting 5 seconds for session to arrive
telegram-bridge_1  | [2021-04-12 12:30:34,819] [WARNING@mau.mx] Didn't find session &lt;session id&gt;, waiting even longer
synapse_1          | 2021-04-12 12:30:34,893 - synapse.handlers.message - 945 - WARNING - POST-8232985 - Denying new eve
nt &lt;FrozenEventV3 event_id='$&lt;event id&gt;', type='m.room.member', state_key='@telegrambot
:&lt;server&gt;'&gt; because 403: You are not invited to this room.
telegram-bridge_1  | [2021-04-12 12:30:34,897] [ERROR@mau.as] Exception in Matrix event handler
telegram-bridge_1  | Traceback (most recent call last):
telegram-bridge_1  |   File "/usr/lib/python3.8/site-packages/mautrix/bridge/matrix.py", line 362, in handle_encrypted
telegram-bridge_1  |     decrypted = await self.e2ee.decrypt(evt, wait_session_timeout=5)
telegram-bridge_1  |   File "/usr/lib/python3.8/site-packages/mautrix/bridge/e2ee.py", line 158, in decrypt
telegram-bridge_1  |     decrypted = await self.crypto.decrypt_megolm_event(evt)
telegram-bridge_1  |   File "/usr/lib/python3.8/site-packages/mautrix/crypto/decrypt_megolm.py", line 40, in decrypt_megolm_event
telegram-bridge_1  |     raise SessionNotFound(evt.content.session_id, evt.content.sender_key)
telegram-bridge_1  | <b>mautrix.errors.crypto.SessionNotFound: Failed to decrypt megolm event: no session with given ID &lt;session id&gt; found</b>
telegram-bridge_1  | 
telegram-bridge_1  | During handling of the above exception, another exception occurred:
telegram-bridge_1  |
telegram-bridge_1  | Traceback (most recent call last):
telegram-bridge_1  |   File "/usr/lib/python3.8/site-packages/mautrix/appservice/api/intent.py", line 335, in ensure_joined
telegram-bridge_1  |     await self.join_room(room_id, max_retries=0)
telegram-bridge_1  |   File "/usr/lib/python3.8/site-packages/mautrix/appservice/api/intent.py", line 83, in wrapper
telegram-bridge_1  |     return await __method(*args, **kwargs)
telegram-bridge_1  |   File "/usr/lib/python3.8/site-packages/mautrix/client/api/rooms.py", line 252, in join_room
telegram-bridge_1  |     content = await self.api.request(Method.POST, Path.join[room_id_or_alias],
telegram-bridge_1  |   File "/usr/lib/python3.8/site-packages/mautrix/api.py", line 238, in request
telegram-bridge_1  |     return await self._send(method, full_url, content, query_params, headers or {})
telegram-bridge_1  |   File "/usr/lib/python3.8/site-packages/mautrix/api.py", line 163, in _send
telegram-bridge_1  |     raise make_request_error(http_status=response.status,
telegram-bridge_1  | mautrix.errors.request.MForbidden: You are not invited to this room.
telegram-bridge_1  |
telegram-bridge_1  | The above exception was the direct cause of the following exception:
telegram-bridge_1  |
telegram-bridge_1  | Traceback (most recent call last):
telegram-bridge_1  |   File "/usr/lib/python3.8/site-packages/mautrix/appservice/as_handler.py", line 178, in try_handle
telegram-bridge_1  |     await handler_func(event)
telegram-bridge_1  |   File "/usr/lib/python3.8/site-packages/mautrix/bridge/matrix.py", line 462, in int_handle_event
telegram-bridge_1  |     await self.handle_encrypted(evt)
telegram-bridge_1  |   File "/usr/lib/python3.8/site-packages/mautrix/bridge/matrix.py", line 364, in handle_encrypted
telegram-bridge_1  |     await self._handle_encrypted_wait(evt, e, wait=10)
telegram-bridge_1  |   File "/usr/lib/python3.8/site-packages/mautrix/bridge/matrix.py", line 384, in _handle_encrypted_wait
telegram-bridge_1  |     event_id = await self.az.intent.send_notice(evt.room_id, msg)
telegram-bridge_1  |   File "/usr/lib/python3.8/site-packages/mautrix/appservice/api/intent.py", line 94, in wrapper
telegram-bridge_1  |     await __self.ensure_joined(room_id)
telegram-bridge_1  |   File "/usr/lib/python3.8/site-packages/mautrix/appservice/api/intent.py", line 339, in ensure_joined
telegram-bridge_1  |     raise IntentError(f"Failed to join room {room_id} as {self.mxid}") from e
telegram-bridge_1  | <b>mautrix.errors.base.IntentError: Failed to join room &lt;room name&gt;:&lt;server&gt; as @telegrambot:&lt;server&gt;</b>
</pre>

Is there any more information I should provide? Would my bridge configuration help?

Thanks in advance!

**Comment by tulir** [comment]
I don't know how you managed to enable encryption without the bot joining the room, but the bridge bot has to be in the room for it to work

**Comment by Craeckie** [comment]
I just noticed that the bridge created an encrypted direct chat even though `encryption.default` is **false**. So this is my first encrypted chat with the bridge, which probably means, that encryption isn't working for me.

So there seem to be two issues:

1. Why did the bridge create an encrypted direct chat, even though the default is false?
2. How do I get encryption to work?

**Comment by Craeckie** [comment]
> I don't know how you managed to enable encryption without the bot joining the room, but the bridge bot has to be in the room for it to work

I'm not able to add the bridge bot, because `!tg set-pl 100 @<my id>` leads to the same error.

**Comment by tulir** [comment]
If you force-enabled encryption in synapse, you have to set it to be on by default in the bridge too

**Comment by Craeckie** [comment]
ah I guess that was the culprit. I had `encryption_enabled_by_default_for_room_type` on `invite`, because it says **default**, not **force**. Then I noticed that it's not possible to create unencrypted DMs anymore, so I set it to `off`, but that encrypted bridge portal was already created. I added matrix-org/synapse#9794 to resolve the misleading naming.

Thanks for that hint, that solves 1).

Regarding 2):
Encryption seems to generally work. I created an encrypted room on Matrix, added the bridge bot and another user and sent a message and it was bridged to Telegram. So I guess it was caused by this combination of forced encryption on synapse and `encryption.default: false` on the bridge.

It would be helpful to add a hint in the configuration file [mautrix_telegram/example-config.yaml#L41](https://github.com/tulir/mautrix-telegram/blob/78fcacf7aa5bb97a85df4b8dcabf34bd7e54c686/mautrix_telegram/example-config.yaml#L41) that is needs to be enabled when `encryption_enabled_by_default_for_room_type` is enabled in synapse.

During this "experiment" I encountered another bug, for which I will create another issue.

Thanks for the help, I guess we can close this.
